src/main/java/seedu/address/logic/Logic.java

```diff
@@ -51,7 +50,7 @@
     ObservableList<Mentor> getFilteredMentorList();
 
     /** Returns the Record of all Commands entered by User */
-    ArrayList<CommandRecord> getCommandHistory() throws AlfredModelHistoryException;
```

Comment:
I threw this exception under the method, so that maybe I can catch it and display the error in gui, or would you prefer other ways of throwing exceptions? 😄

Reply:
Yeah but the thing is that getCommandHistory doesn't throw the exception at all. In the changes you introduced previously, I think the reason it threw this exception is because the constructor for CommandRecord threw the exception. I edited the CommandRecord class with the getEndPoints() static methods to work around the exception throwing.